pkg/agent/cniserver/ipam/ipam_service.go

```diff
@@ -83,6 +89,11 @@ func ExecIPAMAdd(cniArgs *cnipb.CniCmdArgs, ipamType string, resultKey string) (
 }
 
 func ExecIPAMDelete(cniArgs *cnipb.CniCmdArgs, ipamType string, resultKey string) error {
+	// Only release IP when handling CNI request from infra container.
```

Comment:
The spec says cniAdd should be idempotent so it's used to get IP on windows, but we don't have cniDel request for non infra container, right? If this code is called by rollback of CNIAdd, I think we should avoid calling it for non-infra container and make it clear that non infra container will only execute read only operation.

Reply:
Make sence.
I will:
- Check if current container is infra container before call CmdDel as failure rollback.
- Remove infra container check in ExecIPAMDelete.

Done.